packages/core/admin/ee/server/services/review-workflows/stages.js

```diff
@@ -146,3 +190,28 @@ function assertAtLeastOneStageRemain(workflowStages, diffStages) {
     throw new ApplicationError('At least one stage must remain in the workflow.');
   }
 }
+
+/**
+ * Find the id of the nearest object in an array that matches a condition.
+ * Used for searching for the nearest stage that is not deleted.
+ * Starts by searching the elements before the index, then the remaining elements in the array.
+ *
+ * @param {Array} array of stages
+ * @param {Number} index the index to start searching from
+ * @param {Function} condition must evaluate to true for the object to be considered a match
+ * @returns {Number}
+ */
+function findNearestMatchingStageID(array, index, condition) {
+  // Start by searching the elements before the index
+  for (let i = index; i >= 0; i -= 1) {
+    if (condition(array[i])) {
+      return array[i].id;
+    }
+  }
+
+  // If no matching element is found before the index,
+  // search the remaining elements in the array
+  const remainingArray = array.slice(index + 1);
+  const nearestObject = remainingArray.filter(condition)[0];
+  return nearestObject.id;
+}
```

Comment:
Nice job, I was playing and see if I could find alternatives, sharing jjust to look for your opinion too.
```js
function findNearestNonDeletedStageId(stages, startIndex, condition) {
  const previousStages = stages.slice(0, startIndex).filter(condition);

   if (previousStages.length) {
    return previousStages.at(-1).id;
   }
  
  const nextStages = stages.slice(startIndex + 1).filter(condition);
	
  return nonDeletedStages[0].id;
}
```

I find the arguments a bit tricky to understand personally, I think it would be easier to follow if it was:
`function findNearestNonDeletedStageId(stages, stageId, deletedIds) {`

wdyt?

Reply:
definitely agree about updating the argument names.

The function name I'm not so sure, as the condition is a parameter I think the functions purpose is a bit more generic than that name. wdyt?

Reply:
True, the function name should be more generic !